packages/backend/src/tokens/request.ts

```diff
@@ -521,7 +521,11 @@ ${error.getFullMessage()}`,
     }
 
     // Multi-domain development sync flow
-    if (authenticateContext.instanceType === 'development' && isRequestEligibleForMultiDomainSync) {
+    if (
+      authenticateContext.instanceType === 'development' &&
+      isRequestEligibleForMultiDomainSync &&
+      !authenticateContext.clerkUrl.searchParams.has(constants.QueryParameters.ClerkSynced)
```

Comment:
The dev flow relies on the `__clerk_synced` query parameter to terminate the sync flow, so I've added it here to ensure the dev flow doesn't infinitely sync.